de.dlr.sc.virsat.cef.swtbot.test/src/de/dlr/sc/virsat/cef/swtbot/test/CefRefreshValueAndStructureTest.java

```diff
@@ -0,0 +1,136 @@
+		final String EXP_SYSTEM_PARAMETERS_MARGIN = "20";
+		final String EXP_SYSTEM_PARAMETERS_UNIT = "Megabyte: MB";
+
+		// set system parameter values
+		rename(systemParameterItem, EXP_SYSTEM_PARAMETERS_NAME);
+		renameField(SystemParameters.PROPERTY_SYSTEMMARGIN, EXP_SYSTEM_PARAMETERS_MARGIN);
+		bot.comboBox().setSelection(EXP_SYSTEM_PARAMETERS_UNIT);
+		save();
+
+		openEditor(systemItem);
+		
+		final int SYSTEM_ITEM_LEVEL = 1;
+		SWTBotTree systemParametersTree = new SWTBotTree((Tree) bot.widget(WidgetMatcherFactory.widgetOfType(Tree.class), 
+				SYSTEM_ITEM_LEVEL));
+		String systemMarginValue = systemParametersTree.
+				getAllItems()[SYSTEM_MARGIN_ROW_INDEX].cell(VALUE_COLUMN_INDEX).toString();
+		String systemMarginUnit = systemParametersTree.
+				getAllItems()[SYSTEM_MARGIN_ROW_INDEX].cell(UNIT_COLUMN_INDEX).toString();
+
+		assertEquals("SP: " + EXP_SYSTEM_PARAMETERS_NAME, systemParameterItem.getText());
+		assertEquals(EXP_SYSTEM_PARAMETERS_MARGIN, systemMarginValue);
+		assertEquals(EXP_SYSTEM_PARAMETERS_UNIT, systemMarginUnit);
+
+		// set mode duration values
+		systemParameterItem.expand();
+		modeDurationItem = systemParameterItem.getNode(0).expand().getNode(0); // get mode duration editor
+
+		final String EXP_MODE_DURATION_NAME = "TestModeDuration";
+		final Double EXP_MODE_DURATION_DEFAULT_VALUE = 50.0;
+		final String EXP_MODE_DURATION_UNIT = "Minute: min";
+		final String EXP_MODE_DURATION_NOTE = "test note";
+
+		openEditor(modeDurationItem);
+		rename(modeDurationItem, EXP_MODE_DURATION_NAME);
+		renameField(Parameter.PROPERTY_DEFAULTVALUE, EXP_MODE_DURATION_DEFAULT_VALUE.toString()); // set default value in mode duration
+		bot.comboBox().setSelection(EXP_MODE_DURATION_UNIT);
+		renameField(Parameter.PROPERTY_NOTE, EXP_MODE_DURATION_NOTE);
+		
+		String modeDurationNote = getPropertyValue(Parameter.PROPERTY_NOTE);
+		assertEquals(EXP_MODE_DURATION_NOTE, modeDurationNote);
+		
+		openEditor(systemItem);
+		
+		String modeDurationName = systemParametersTree.
+				getAllItems()[MODE_DURATION_ROW_INDEX].cell(NAME_COLUMN_INDEX);
+		String modeDurationDefaultValue = systemParametersTree.
+				getAllItems()[MODE_DURATION_ROW_INDEX].cell(VALUE_COLUMN_INDEX);
+		String modeDurationUnit = systemParametersTree.
+				getAllItems()[MODE_DURATION_ROW_INDEX].cell(UNIT_COLUMN_INDEX);
+		
+		final Double DELTA_DOUBLE = 0.00001;
+		
+		assertEquals(EXP_MODE_DURATION_NAME, modeDurationName);
+		assertEquals(EXP_MODE_DURATION_DEFAULT_VALUE, Double.parseDouble(modeDurationDefaultValue), DELTA_DOUBLE);
+		assertEquals(EXP_MODE_DURATION_UNIT, modeDurationUnit);
+	}
+	
+	@Test
+	public void refreshTreeTablePropertyStructureTest() {
+		systemParameterItem = addElement(SystemParameters.class, conceptCef, systemItem);
+		SWTBotTree systemParametersTree = new SWTBotTree((Tree) bot.widget(WidgetMatcherFactory.widgetOfType(Tree.class), 
+				SYSTEM_ITEM_LEVEL));
+		systemParametersTree.getAllItems()[MODE_DURATION_ROW_INDEX].expand();
```

Comment:
`systemParametersTree.getAllItems()[MODE_DURATION_ROW_INDEX]` this expression seems to appear a couple of times. Maybe we can put the item into a variable?